### ISSUE TITLE
Encoder credit learning rate is 1000x too slow to shape representations

### ISSUE THREAD

**Issue by koraytaylan** [koraytaylan/xagent#91]
## Problem

The encoder weight update in pass 7 (`coop_learn_and_store`) applies an additional `0.001` scaling factor on top of the already-weak credit signal, making encoder learning effectively frozen. The encoder stays near its random Xavier initialization and cannot learn to represent actionable spatial information like "food is to my left."

## Location

`crates/xagent-brain/src/shaders/kernel/brain_tick.wgsl:629` — the `0.001` multiplier
`crates/xagent-brain/src/shaders/kernel/brain_tick.wgsl:626-636` — encoder credit update block

### The update rule

```wgsl
if (tid < DIM) {
    let credit_d = decision_buf[d_base + DIM + tid];
    if (abs(credit_d) >= 1e-6) {
        let scale = learning_rate * credit_d * 0.001;  // ← 0.001 multiplier
        for (var j: u32 = 0u; j < FEATURE_COUNT; j = j + 1u) {
            var w = brain_state[b + O_ENC_WEIGHTS + j * DIM + tid] + scale * s_features[j];
            w = clamp(w, -2.0, 2.0);
            brain_state[b + O_ENC_WEIGHTS + j * DIM + tid] = w;
        }
    }
}
```

The effective encoder learning rate is `learning_rate × credit × 0.001`. With:
- `learning_rate`: a heritable config value (typical ~0.01-0.1)
- `credit`: already weak due to DEADZONE and noise issues (typical ~0.01 when it fires)
- `0.001`: the additional constant

Effective rate: `~0.1 × ~0.01 × 0.001 = 0.000001` per feature per dimension per tick.

### Why encoder shaping matters

The encoder is a single linear layer: `265 features → 32 dimensions` with `tanh`. With Xavier initialization (`scale = 1/√265 ≈ 0.062`), this is a random projection. A random 265→32 projection has some useful properties (Johnson-Lindenstrauss), but it doesn't preserve the specific spatial structure needed for behavior:

- "Food is in ray column 3" vs "Food is in ray column 6" should map to different encoded directions
- "Food is close (low depth)" vs "Food is far (high depth)" should be distinguishable
- "Danger biome color in lower-left" vs "safe biome color" should separate

A random projection mixes these signals unpredictably. The encoder MUST be shaped by credit to align its representation with behavioral relevance. But at `0.000001` per tick, this shaping is negligible.

### Quantitative estimate

Encoder weight range: `[-2.0, 2.0]` (clamped). Initial scale: `~0.062`. To meaningfully reshape the projection (change weights by ~0.1 from their initial values):

```
ticks_needed = 0.1 / 0.000001 = 100,000 ticks of active credit
```

But credit only fires for ~5 ticks per food encounter (see companion DEADZONE issue), so:

```
food_encounters = 100,000 / 5 = 20,000 encounters
```

At ~1 encounter per 200 brain ticks: **4,000,000 brain ticks** to begin shaping the encoder. This is orders of magnitude beyond "thousands of iterations."

### The cascade effect

Without a shaped encoder:
1. The 32-dim encoded state doesn't reliably distinguish "food left" from "food right"
2. Credit assignment can't learn "turn left when food is left" because the state vectors for those two situations are similar
3. Even if credit fires correctly, the weight updates don't correlate with the useful spatial dimension
4. Policy weights drift randomly instead of aligning with food-relevant directions

## Suggested direction

1. **Remove or significantly reduce the `0.001` multiplier** — a factor of `0.1` instead of `0.001` would make encoder learning 100× faster while still being slower than action weight learning (appropriate since the encoder should be more stable)
2. **Consider separate learning rate for encoder** — rather than scaling the action credit, use a dedicated encoder learning rate that can be tuned independently
3. **Weight clamp `[-2.0, 2.0]` is the safety net** — it already prevents runaway encoder weights, so the extreme conservatism of `0.001` is unnecessary

## Parent issue

Part of #13 — Emergent food-seeking and danger-avoidance behaviors fail to develop
